src/Atmos/Model/subsidence.jl

```diff
@@ -0,0 +1,16 @@
+#### Subsidence model
+export AbstractSubsidence,
+       NoSubsidence,
+       ConstantSubsidence,
+       subsidence_velocity
+
+abstract type AbstractSubsidence{FT<:AbstractFloat} end
+
+struct NoSubsidence{FT} <: AbstractSubsidence{FT} end
+
+struct ConstantSubsidence{FT} <: AbstractSubsidence{FT}
+  D::FT
+end
+
+subsidence_velocity(::NoSubsidence{FT}, z::FT) where {FT} = SVector{FT}(0,0,0)
```

Comment:
I'd define this as a scalar (vertical) velocity here, defined positive downward. Fewer zeros to carry around, and more in line how people use the term.